doc/develop/west/sign.rst

```diff
@@ -94,3 +94,54 @@ file an issue for adding support.
 
 .. _imgtool:
    https://pypi.org/project/imgtool/
+
+
+ZEPHYR_WEST_SIGN_CONFIG file
+****************************
+
+Signing an image as described above involves a number of "wrapper" scripts: ``west build``
+invokes ``cmake`` and ``ninja`` which invokes ``west sign`` which invokes ``imgtool`` or
+`rimage`_. As long as the default options are appropriate these indirections are not a
+problem. On other hand, passing ``imgtool`` or ``rimage`` command line parameters through
+all these layers causes issues typical when stacking multiple layers that don't
+abstract anything.
+
+First, this usually requires boilerplate code in each wrapper. Some wrappers may have
+issues escaping whitespace or other special characters. Replicating the invocation of a
+lower layer like ``west sign`` directly to troubleshoot some build issue requires scanning
+verbose build logs to find which exact options were used. Last and worst: each brand
+new option is impossible to use until more boilerplate code has been added.
+
+To avoid these issues, ``west sign`` supports setting ``imgtool`` and ``rimage`` parameters in
+both the usual ``west`` configuration files and in a dedicated ``.toml`` configuration
+file:
+
+.. code-block:: ini
+
+   # Same section name as in west config
+   [sign]
+   # As opposed to west config files, TOML strings must be quoted
+   tool = "rimage"
+   tool-path = "/home/me/SOF/build-rimage/rimage"
+
+   # west config cannot do this because .ini parsers treat everything as a string
+   tool-extra-args = [ "-i", "4", "-k", "keys/key argument with space.pem" ]
```

Comment:
Is the concern here avoiding a need to split a string according to platform-specific whitespace rules? We have build.cmake-args already as a string, e.g.

Reply:
Yes.

```shell
west config sign.tool-extra-args '-i 4 -k "sof/keys/some private key.pem"'

west config sign.tool-extra-args
-i 4 -k "sof/keys/some private key.pem" # looks good so far but...

west sign -t rimage

rimage -i 4 -k '"sof/keys/some'  private 'key.pem"'
```

This is typical, from http://mywiki.wooledge.org/BashGuide/Arrays :
> The only safe way to represent multiple string elements [in Bash] is through the use of arrays.

I'm not aware of any `.ini`-compatible solution.

> We have build.cmake-args already as a string, e.g.

I don't understand sorry. I looked a https://docs.zephyrproject.org/3.1.0/develop/west/manifest.html and don't see what you're referring to.

Reply:
There is also the issue that `west config` is persistent, not build-directory specific and storing unrelated data. The new `sign config.toml` files can be versioned/copied/deleted independently and separately for each build directory.